app/cloud_tasks/cloud_task_publishers.py

```diff
@@ -0,0 +1,58 @@
+from google import auth
+from google.cloud.tasks_v2 import CloudTasksClient, HttpMethod
+from google.cloud.tasks_v2.types.task import Task
+from structlog import get_logger
+
+from app.settings import SUBMISSION_CONFIRMATION_CLOUD_FUNCTION_NAME
+
+from .exceptions import CloudTaskCreationFailed
+
+logger = get_logger(__name__)
+
+
+class CloudTaskPublisher:
+    def __init__(self):
+        self._client = CloudTasksClient()
+
+        _, self._project_id = auth.default()
+
+    def _create(self, payload: bytes, queue_name: str) -> Task:
+        logger.info("creating cloud task")
+
+        self._parent = self._client.queue_path(
+            self._project_id, "europe-west2", queue_name
+        )
+
+        service_account_email = (
+            f"cloud-functions@{self._project_id}.iam.gserviceaccount.com"
+        )
+
+        url = f"https://europe-west2-{self._project_id}.cloudfunctions.net/{SUBMISSION_CONFIRMATION_CLOUD_FUNCTION_NAME}"
```

Comment:
We need to validate `SUBMISSION_CONFIRMATION_CLOUD_FUNCTION_NAME` exists. Given we tie having cloud functions to a task client, this should be checked when we initialise the `CloudTaskPublisher`, alternatively given you can have multiple queues, you could have multiple functions for each queue, which means this should be passed in like the queue nam.

@berroar @ajmaddaford thoughts?